POS/Morph annotation empty with trained Transformer model

### How to reproduce the behaviour
<!-- Include a code example or the steps that led to the problem. Please try to be as specific as possible. -->
I have trained a Swedish Transformer model on UD-Treebank Talbanken using mainly the [Quickstart Config](https://nightly.spacy.io/usage/training#quickstart) from the Spacy website. I only added `morphologizer` and `lemmatizer` to the model. So the full pipeline looks like this:

`['transformer', 'tagger', 'morphologizer', 'lemmatizer', 'parser']
`

 With normal sentences everything seems to work fine, but when very short sequences _(such as `nlp('Privat')`)_ are passed through the model the POS and Morph annotation can be missing for some tokens. 

Here are some examples with first the text sequence and then a list with respective POS and Morph annotations returned by `token.pos_` and `token.morph`:

```
Privat [''] []
A-Ö ['PROPN', 'SYM', ''] [Case=Nom, , ]
A [''] []
B [''] []
C ['NOUN'] [Abbr=Yes]
D [''] []
E [''] []
F [''] []
G ['PROPN'] [Case=Nom]
H [''] []
I ['ADP'] []
J [''] []
K [''] []
L ['ADP'] []
M [''] []
N ['PROPN'] [Case=Nom]
O ['PUNCT'] []
P [''] []
Q [''] []
R [''] []
S ['ADP'] []
T [''] []
U [''] []
V [''] []
W [''] []
X [''] []
Y [''] []
Z [''] []
Å ['INTJ'] []
Ä [''] []
Ö [''] []
```

I also trained a model on both the UD-Talbanken and UD-Lines corpora. That model has the issue as well, but it seems to occur not that frequently:

```
Privat ['PUNCT'] []
A ['NOUN'] [Case=Nom|Definite=Ind|Gender=Neut|Number=Sing]
-Ö ['SYM', ''] [, ]
A ['ADV'] []
B ['PUNCT'] []
C ['PUNCT'] []
D ['PUNCT'] []
E ['PUNCT'] []
F ['PUNCT'] []
G ['PUNCT'] []
H ['PUNCT'] []
I ['PUNCT'] []
J ['PUNCT'] []
K ['PUNCT'] []
L ['PUNCT'] []
M [''] []
N ['PUNCT'] []
O ['PUNCT'] []
P ['PUNCT'] []
Q ['PUNCT'] []
R ['PUNCT'] []
S ['PUNCT'] []
T ['PUNCT'] []
U ['PUNCT'] []
V ['PUNCT'] []
W ['PUNCT'] []
X ['NOUN'] []
Y ['PUNCT'] []
Z ['PUNCT'] []
Å ['INTJ'] []
Ä ['ADV'] []
Ö ['PUNCT'] []
```
_(Please note that the list containing the `morph` annotation actually contains a `spacy.tokens.morphanalysis.MorphAnalysis
` object where the `__str__/__repr__` method returns an empty string.)_


`TAG` and `dep` annotations _seem_ to be okay from what I can see. 

Is there maybe some threshold for certainty/entropy that prevents predicting POS tags for uncertain cases? 

If that is the case, than I would rather prefer uncertain tags over no tags, because currently I'm getting an error message from Spacy `Matcher`:
```
ValueError: [E155] The pipeline needs to include a morphologizer in order to use Matcher or PhraseMatcher with the attribute POS. Try using `nlp()` instead of `nlp.make_doc()` or `list(nlp.pipe())` instead of `list(nlp.tokenizer.pipe())`.
```

Please let me know if you need additional information. Thanks!


### Your Environment
<!-- Include details of your environment. If you're using spaCy 1.7+, you can also type `python -m spacy info --markdown` and copy-paste the result here.-->
* Operating System:
`Ubuntu 20.04.1 LTS
`
* Python Version Used:
```
Python 3.8.3 (default, May 19 2020, 18:47:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
```
* spaCy Version Used: 
`3.0.0rc2`

No, I trained until the training script stopped by itself. The notebook shows a later test that I aborted.

Here is an older training output also trained on Talbanken. The training script is the same, I only used different `lemma_exc` data.

The model performance itself, also on other data is also not all too bad. POS and DEP labels look reasonable. 

However the [UD-Talbanken](https://universaldependencies.org/treebanks/sv_talbanken/index.html) treebank is not very large and the problem was less common when I added the [UD-Lines](https://universaldependencies.org/treebanks/sv_lines/index.html) treebank, which doubled the amount of training data. So there could be a correlation with the amount of training data. 

I don't know how the morph model works, but could it be that even after training some morphs remained empty?


```
2020-11-02 16:03:39.495695: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
ℹ Using GPU: 0

=========================== Initializing pipeline ===========================
Set up nlp object from config
Pipeline: ['transformer', 'tagger', 'morphologizer', 'lemmatizer', 'parser']
Created vocabulary
Finished initializing nlp object
Initialized pipeline components: ['transformer', 'tagger', 'morphologizer', 'lemmatizer', 'parser']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['transformer', 'tagger', 'morphologizer', 'lemmatizer',
'parser']
ℹ Initial learn rate: 0.0
E    #       LOSS TRANS...  LOSS TAGGER  LOSS MORPH...  LOSS PARSER  TAG_ACC  POS_ACC  MORPH_ACC  LEMMA_ACC  DEP_UAS  DEP_LAS  SENTS_F  SCORE 
---  ------  -------------  -----------  -------------  -----------  -------  -------  ---------  ---------  -------  -------  -------  ------
  0       0        2649.04      1292.14        1295.62      1904.61     5.65     0.00      35.63      62.41    19.15     0.73     0.00    0.24
  5     200      414165.46    349024.56      351501.46    314286.58    53.23    87.09      58.23      83.65    81.44    74.00    92.64    0.70
 11     400      101147.58    163143.18      171550.76     81703.36    93.33    98.01      93.74      87.00    89.05    85.09    96.27    0.89
 16     600       44678.66     13504.81       16589.89     31378.16    97.37    98.66      97.37      87.01    91.48    88.54    97.53    0.91
 21     800       19743.16      3001.88        4720.89     11772.16    97.62    98.76      97.61      87.06    92.69    89.84    97.82    0.92
 27    1000        8006.59      1393.63        2429.80      4405.85    97.76    98.82      97.87      87.01    92.39    89.73    97.63    0.92
 32    1200        4936.63       643.29        1296.42      2246.22    97.73    98.85      97.84      87.04    92.40    89.82    97.72    0.92
 38    1400        4526.94       384.91         717.74      1607.24    97.68    98.86      97.90      87.03    92.11    89.56    97.92    0.92
 43    1600        2703.55       255.31         425.99       984.58    97.82    98.89      97.99      87.08    92.47    89.99    97.33    0.92
 49    1800        2387.74       189.04         294.65       831.95    97.64    98.89      97.94      87.08    92.16    89.71    97.34    0.92
 54    2000        2428.07       102.84         140.88       803.04    97.86    98.87      97.95      87.04    92.40    89.92    96.95    0.92
 60    2200        1544.83       109.84         136.95       519.11    97.73    98.85      97.92      87.14    92.58    90.12    97.04    0.92
 65    2400        1946.26       105.30         133.34       605.98    97.92    98.86      98.06      87.06    92.35    89.76    96.95    0.92
 71    2600        1434.08       104.13         129.62       394.13    97.77    98.80      97.94      87.11    91.95    89.32    97.83    0.92
 76    2800        1201.01        94.52          90.80       323.93    97.74    98.94      97.94      87.05    92.34    89.85    97.24    0.92
 82    3000        1499.74        90.62          90.63       343.52    97.69    98.83      97.92      87.13    92.27    89.75    97.92    0.92
 87    3200        1562.12        92.26          91.54       350.25    97.82    98.82      98.01      87.11    92.32    89.92    97.44    0.92
 92    3400        1219.65        97.56          96.69       284.64    97.74    98.94      97.87      87.05    92.79    90.14    98.22    0.92
 98    3600        1179.11        66.65          98.81       278.46    97.81    98.87      97.94      87.05    92.55    89.97    97.24    0.92
103    3800        1515.70        57.46          58.29       321.78    97.83    98.82      97.99      87.03    92.37    89.74    98.12    0.92
109    4000        1071.01        36.30          44.84       208.29    97.83    98.89      97.98      87.07    92.36    89.86    97.83    0.92
114    4200         986.62        78.65          75.49       204.65    97.81    98.84      97.97      87.04    92.41    89.90    97.83    0.92
120    4400        1073.68        58.29          70.97       205.42    97.72    98.84      97.93      87.08    92.40    90.00    97.63    0.92
125    4600         798.90        56.53          63.53       156.42    97.71    98.94      97.97      87.05    92.64    90.04    98.52    0.92
131    4800         688.09        29.01          32.98       152.94    97.75    98.86      97.91      87.09    92.63    90.16    98.22    0.92
136    5000         979.26        26.85          30.70       192.99    97.87    98.93      98.05      87.03    92.53    89.97    98.52    0.92
✔ Saved pipeline to output directory
output/model-last
```

--- a/spacy/pipeline/morphologizer.pyx
+++ b/spacy/pipeline/morphologizer.pyx
@@ -92,9 +92,6 @@ class Morphologizer(Tagger):
         # 2) labels_pos stores a mapping from morph+POS->POS
         cfg = {"labels_morph": labels_morph or {}, "labels_pos": labels_pos or {}}
         self.cfg = dict(sorted(cfg.items()))
-        # add mappings for empty morph
-        self.cfg["labels_morph"][Morphology.EMPTY_MORPH] = Morphology.EMPTY_MORPH
-        self.cfg["labels_pos"][Morphology.EMPTY_MORPH] = POS_IDS[""]
 
     @property
     def labels(self):
@@ -201,8 +198,8 @@ class Morphologizer(Tagger):
                 doc_tag_ids = doc_tag_ids.get()
             for j, tag_id in enumerate(doc_tag_ids):
                 morph = self.labels[tag_id]
-                doc.c[j].morph = self.vocab.morphology.add(self.cfg["labels_morph"][morph])
-                doc.c[j].pos = self.cfg["labels_pos"][morph]
+                doc.c[j].morph = self.vocab.morphology.add(self.cfg["labels_morph"].get(morph, 0))
+                doc.c[j].pos = self.cfg["labels_pos"].get(morph, 0)
 
     def get_loss(self, examples, scores):
         """Find the loss and gradient of loss for the batch of documents and
@@ -228,12 +225,12 @@ class Morphologizer(Tagger):
                 # doesn't, so if either is None, treat both as None here so that
                 # truths doesn't end up with an unknown morph+POS combination
                 if pos is None or morph is None:
-                    pos = None
-                    morph = None
-                label_dict = Morphology.feats_to_dict(morph)
-                if pos:
-                    label_dict[self.POS_FEAT] = pos
-                label = self.vocab.strings[self.vocab.morphology.add(label_dict)]
+                    label = None
+                else:
+                    label_dict = Morphology.feats_to_dict(morph)
+                    if pos:
+                        label_dict[self.POS_FEAT] = pos
+                    label = self.vocab.strings[self.vocab.morphology.add(label_dict)]
                 eg_truths.append(label)
             truths.append(eg_truths)
         d_scores, loss = loss_func(scores, truths)

--- a/spacy/tests/pipeline/test_morphologizer.py
+++ b/spacy/tests/pipeline/test_morphologizer.py
@@ -116,3 +116,23 @@ def test_overfitting_IO():
     no_batch_deps = [doc.to_array([MORPH]) for doc in [nlp(text) for text in texts]]
     assert_equal(batch_deps_1, batch_deps_2)
     assert_equal(batch_deps_1, no_batch_deps)
+
+    # Test without POS
+    nlp.remove_pipe("morphologizer")
+    nlp.add_pipe("morphologizer")
+    for example in train_examples:
+        for token in example.reference:
+            token.pos_ = ""
+    optimizer = nlp.initialize(get_examples=lambda: train_examples)
+    for i in range(50):
+        losses = {}
+        nlp.update(train_examples, sgd=optimizer, losses=losses)
+    assert losses["morphologizer"] < 0.00001
+
+    # Test the trained model
+    test_text = "I like blue ham"
+    doc = nlp(test_text)
+    gold_morphs = ["Feat=N", "Feat=V", "", ""]
+    gold_pos_tags = ["", "", "", ""]
+    assert [str(t.morph) for t in doc] == gold_morphs
+    assert [t.pos_ for t in doc] == gold_pos_tags

